Lib/asyncio/selector_events.py

```diff
@@ -555,20 +555,19 @@ async def sock_accept(self, sock):
         if self._debug and sock.gettimeout() != 0:
             raise ValueError("the socket must be non-blocking")
         fut = self.create_future()
-        self._sock_accept(fut, False, sock)
+        self._sock_accept(fut, sock)
         return await fut
 
-    def _sock_accept(self, fut, registered, sock):
+    def _sock_accept(self, fut, sock):
         fd = sock.fileno()
-        if registered:
-            self.remove_reader(fd)
-        if fut.done():
-            return
         try:
             conn, address = sock.accept()
             conn.setblocking(False)
         except (BlockingIOError, InterruptedError):
-            self.add_reader(fd, self._sock_accept, fut, True, sock)
+            self._ensure_fd_no_transport(fd)
+            handle = self.add_reader(fd, self._sock_accept, fut, sock)
```

Comment:
should be `self._add_reader`

Reply:
Fixed.

Reply:
I'm curious why this wasn't caught by tests...

Reply:
What exactly? The unprefixed `add_reader()` just calls `self._ensure_fd_no_transport(fd)` before calling `self._add_reader()`, so I could have saved one line by using that and deleting the explicit call to that.

Reply:
There's a fundamental difference between them. One returns None, the other returns something else that you appear to use in your code.

Reply:
Right, I see the problem now. Oddly, if I comment out the line that adds the callback, the test **does** trigger the failure.

Reply:
I just tried changing it back to unprefixed `add_reader()` and got a bunch of timeout/cancelled errors in other tests.

Reply:
Awesome! The question is now closed then :)